src/main/java/edu/wisc/my/personalizedredirection/controller/PersonalizedRedirectionController.java

```diff
@@ -79,7 +82,13 @@ public void setRedirectionService(IRedirectionService redirectionService) {
             logger.error("Issues happened while trying to generate custom link",
                     e);
             response.setStatus(HttpServletResponse.SC_NOT_FOUND);
+            hasError = true;
          }
+
+         if(hasError){
+            response.sendRedirect(errorURL);
```

Comment:
Without the explicit redirect, it just barfs and does this.

![image](https://cloud.githubusercontent.com/assets/1079433/23320003/95ea35b4-fa9e-11e6-9832-73c096102d76.png)